flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateFunction.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.flink.sql.parser.error.SqlValidateException;
+
+import org.apache.calcite.sql.SqlCharStringLiteral;
+import org.apache.calcite.sql.SqlCreate;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import javax.annotation.Nonnull;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * CREATE FUNCTION DDL sql call.
+ */
+public class SqlCreateFunction extends SqlCreate implements ExtendedSqlNode {
+
+	public static final SqlSpecialOperator OPERATOR = new SqlSpecialOperator("CREATE FUNCTION", SqlKind.CREATE_FUNCTION);
+
+	private final SqlIdentifier functionName;
+
+	private final SqlCharStringLiteral functionClassName;
+
+	private final String functionLanguage;
+
+	private final boolean hasTemporary;
+
+	private final boolean isSystemFunction;
+
+	public SqlCreateFunction(
+		SqlParserPos pos,
+		SqlIdentifier functionName,
+		SqlCharStringLiteral functionClassName,
+		String functionLanguage,
+		boolean ifNotExists,
+		boolean hasTemporary,
+		boolean isSystemFunction) {
+		super(OPERATOR, pos, false, ifNotExists);
+		this.functionName = requireNonNull(functionName);
+		this.functionClassName = requireNonNull(functionClassName);
+		this.isSystemFunction = requireNonNull(isSystemFunction);
+		this.hasTemporary = hasTemporary;
+		this.functionLanguage = functionLanguage;
+	}
+
+	@Override
+	public SqlOperator getOperator() {
+		return OPERATOR;
+	}
+
+	@Nonnull
+	@Override
+	public List<SqlNode> getOperandList() {
+		return ImmutableNullableList.of(functionName, functionClassName);
+	}
+
+	@Override
+	public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+		writer.keyword("CREATE");
+		if (hasTemporary) {
+			writer.keyword("TEMPORARY");
+		}
+		if (isSystemFunction) {
+			writer.keyword("SYSTEM");
+		}
+		writer.keyword("FUNCTION");
+		if (ifNotExists) {
+			writer.keyword("IF NOT EXISTS");
+		}
+		functionName.unparse(writer, leftPrec, rightPrec);
+		writer.keyword("AS");
+		functionClassName.unparse(writer, leftPrec, rightPrec);
+		if (functionLanguage != null) {
+			writer.keyword("LANGUAGE");
+			writer.keyword(functionLanguage);
+		}
+	}
+
+	@Override
+	public void validate() throws SqlValidateException {
+		// no-op
+	}
+
+	public boolean isIfNotExists() {
+		return ifNotExists;
+	}
+
+	public boolean isSystemFunction() {
+		return isSystemFunction;
+	}
+
+	public SqlIdentifier getFunctionName() {
+		return this.functionName;
+	}
+
+	public SqlCharStringLiteral getFunctionClassName() {
+		return this.functionClassName;
+	}
+
+	public String getFunctionLanguage() {
+		return this.functionLanguage;
+	}
+
+	public String[] fullFunctionName() {
```

Comment:
Not sure why we need this, and it doesn't necessarily return "full" name.

Reply:
Simplified it as getFunctionName()